nilearn/regions/region_extractor.py

```diff
@@ -169,6 +170,13 @@ def connected_regions(maps_img, min_region_size=1350,
         index_of_each_map.extend([index] * len(regions))
         all_regions_imgs.extend(regions)
 
+    if len(all_regions_imgs) == 0:
+        warnings.warn("No regions are found to be extracted. Check if the "
+                      "given 'maps_img' is not empty. "
+                      "Returning image as it is.", stacklevel=2)
+        input_img = new_img_like(maps_img, maps_img.get_data())
```

Comment:
It is not valid as you are absolutely not returning what is described in the doc. Plus, the length of `regions_extracted_img` and `index_of_each_map` should be the same, which is not the case with your current code.

I am not against returning special values in case of error to keep the pipeline running, but what you do here may be mlisleading for people.